### PR TITLE
[GST]: make dash mimetype support configurable

### PR DESCRIPTION
**Pull request by emutavchi**
Similar to HLS.

### CODE REVIEW

**Comment by philn** [comment]
Do you have examples of sites relying on DASH?

The HLS opt-out was put in place because usually the WPE UA was detected as Safari, which supports HLS, but hlsdemux was so buggy. These websites provide MSE as fallback, Twitter and TED for instance.

So what's the use-case for this PR?

File: Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp [hunk 1]
```diff
@@ -475,6 +474,11 @@ void GStreamerRegistryScanner::initializeDecoders(const GStreamerRegistryScanner
             mapping.append({ ElementFactories::Type::Demuxer, "application/x-hls", { "application/vnd.apple.mpegurl"_s, "application/x-mpegurl"_s }, { } });
     }
 
+    if (const char* hlsSupport = g_getenv("WEBKIT_GST_ENABLE_DASH_SUPPORT")) {
```
Comment (jacek-skiba-red) [inline Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp]: please change hlsSupport to dashSupport

**Comment by emutavchi** [comment]
Hi @philn. Paramount+ is first web app where we noticed the problem. It uses Shaka player, that seem to prefer native player when user agent can play it. So, I suppose other web apps that use Shaka could be affected as well.

File: Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp (hunk 1 above)
Comment (emutavchi) [reply Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp]: done

**Comment by jacek-skiba-red** [comment]
> Paramount+ is first web app where we noticed the problem. It uses Shaka player

specifying: from my understanding, Paramount+ has internal video player (Avia) which is based on Shaka

**Comment by philn** [comment]
I'm not opposing this change, but you might be able to achieve the same result by setting this env var: `GST_PLUGIN_FEATURE_RANK=dashdemux:0`.

**Comment by emutavchi** [comment]
```
I'm not opposing this change, but you might be able to achieve the same result by setting this env var: GST_PLUGIN_FEATURE_RANK=dashdemux:0.
```
That will require a change in environment configuration for every browser launcher (we do have several different launchers/configurations). It is an error prone approach in my opinion ...

I'd prefer to make a change in single place. And have dashdemux support enabled only in cases where it is needed.

**Comment by philn** [comment]
Alright then, can you open a PR for this upstream?

**Comment by modeveci** [comment]
@philn the decision of a PR to be upstreamed handled by us. Could you please check with Xavier Castano? Thank you!

**Comment by philn** [comment]
I'll take care of it then, thanks!

**Comment by philn** [comment]
https://github.com/WebKit/WebKit/pull/14535

**Review by modeveci — APPROVED** [review]
@philn or @calvaris can we merge this to downstream wpe-2.38? thank you!

**Comment by philn** [comment]
done.

**Pull request closed without merging**